lib/ddtrace/contrib/active_model_serializers/patcher.rb

```diff
@@ -0,0 +1,78 @@
+module Datadog
+  module Contrib
+    module ActiveModelSerializers
+      # Provides instrumentation for ActiveModelSerializers through ActiveSupport instrumentation signals
+      module Patcher
+        include Base
+        VERSION_REQUIRED = Gem::Version.new('0.9.0')
+        EVENT_NAME_10 = 'render.active_model_serializers'.freeze
+        EVENT_NAME_09 = '!serialize.active_model_serializers'.freeze
+        register_as :active_model_serializers
+        option :tracer, default: Datadog.tracer
+        option :service_name, default: 'active_model_serializers'
+
+        class << self
+          def patch
+            return patched? if patched? || !compatible?
+
+            ::ActiveSupport::Notifications.subscribe(event_name, self)
+
+            configuration[:tracer].set_service_info(
+              configuration[:service_name],
+              'active_model_serializers',
+              Ext::AppTypes::WEB
+            )
+
+            @patched = true
+          end
+
+          def patched?
+            return @patched if defined?(@patched)
+            @patched = false
+          end
+
+          def call(_, started, finished, _, payload)
+            ensure_clean_context!
+
+            span = configuration[:tracer].trace(payload[:serializer])
```

Comment:
I think our API is a little misleading here, but the argument for `trace` should be a `name`, which is a generic name for a kind of span e.g. `active_model.serializer.render`, not a `resource`, which is a name that uniquely identifies what category of user data this belongs to e.g. `FooSerializer`. (I think that's what this payload represents.)

I would change this argument to a fixed constant of a name. I think the reason you might not be seeing traces/spans show up might be because `resource` is `nil`, causing the spans to be filtered out. If you give `resource` a value, that might fix it.

Reply:
@delner sorry for taking a bit to get back about this, I was out of town over the weekend.

I tried that, but the app is still not tracking in APM when I try to use this fork. It's the entire app that won't track, not just the active model serializers span. There are no errors in the logs, and the debug info looks like everything should be working.

Reply:
@skisulli Sorry for the late reply! This sometimes happens when there are unfinished spans in the trace. If that happens, it will discard the whole trace.

You can enable debug tracing to see if there are any messages/warnings indicating this might be the case:

```ruby
Datadog.configure do |c|
  c.tracer debug: true
end
```

Want to give that a shot and see if it prints out any useful output?